src/main/java/appeng/block/AEBaseTileBlock.java

```diff
@@ -271,87 +271,93 @@ public void onBlockPlacedBy( final World w, final BlockPos pos, final IBlockStat
 	@Override
 	public boolean onBlockActivated( final World w, final BlockPos pos, final IBlockState state, final EntityPlayer player, final EnumHand hand, final @Nullable ItemStack heldItem, final EnumFacing side, final float hitX, final float hitY, final float hitZ )
 	{
-		if( player != null )
+		if( player != null && heldItem != null )
 		{
-			if( heldItem != null )
+			if( Platform.isWrench( player, heldItem, pos ) && player.isSneaking() )
 			{
-				if( Platform.isWrench( player, heldItem, pos ) && player.isSneaking() )
+				final IBlockState blockState = w.getBlockState( pos );
+				final Block block = blockState.getBlock();
+
+				if( block == null )
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Redundant nullcheck of block, which is known to be non-null in appeng.block.AEBaseTileBlock.onBlockActivated(World, BlockPos, IBlockState, EntityPlayer, EnumHand, ItemStack, EnumFacing, float, float, float) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=findbugs%3ARCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE)